humility-core/src/core.rs

```diff
@@ -1514,6 +1514,31 @@ fn get_usb_probe(index: Option<usize>) -> Result<probe_rs::DebugProbeInfo> {
     }
 }
 
+fn open_probe_from_selector(
+    selector: probe_rs::DebugProbeSelector,
+) -> Result<probe_rs::Probe> {
+    let res = probe_rs::Probe::open(selector.clone());
+
+    if let Err(probe_rs::DebugProbeError::ProbeCouldNotBeCreated(
+        probe_rs::ProbeCreationError::NotFound,
+    )) = res
+    {
+        if selector.serial_number.is_some() {
+            bail!(
+                "Could not find probe {}. Because a serial number is present, \
```

Comment:
take it or leave it: I might put a line break between "Could not find probe  {}" and the  rest of the help  text?

Reply:
yes! i didn't like how long the line was but i couldn't quite figure out how i'd want to break it. lemme do that.

Reply:
much more legible with https://github.com/oxidecomputer/humility/pull/504/commits/6f29873972f4affc381338fac73cac24b826989d:

```
humility sensors failed: Could not find probe 0483:374e:002900184741500520383733. 

Because a serial number is present, this may be due to not running humility with permission to read USB device serial numbers; if not root already, run again as root?
```